filesystem/ios/Plugin/Filesystem.swift

```diff
@@ -0,0 +1,156 @@
+import Foundation
+
+@objc public class Filesystem: NSObject {
+
+    public enum FilesystemError: String, Error {
+        case noParentFolder = "Parent folder doesn't exist"
+        case noSave = "Unable to save file"
+        case failEncode = "Unable to encode data to utf-8"
+        case noAppend = "Unable to append file"
+        case notEmpty = "Folder is not empty"
+    }
+
+    @objc public func readFile(at fileUrl: URL, with encoding: String?) throws -> String {
```

Comment:
Are you sure this will work? It used to be that a Swift function cannot be exposed to Obj-C if it both throws and returns a value (since the Obj-C pattern is to return a `BOOL` for success and use a pointer for the result and a handle for the `NSError`). I don't know if that's changed recently, though. It might be better to not expose this class to Obj-C at all if the methods can't be cleanly mapped.

Reply:
I works in Swift, didn't try on Objective-C